includes/Core/Util/Site_Health_Status.php

```diff
@@ -0,0 +1,284 @@
+				__( 'There was an error while trying to get the status, please try again later.', 'google-site-kit' )
+			);
+
+			return $result;
+		}
+
+		$response = wp_remote_retrieve_body( $response );
+
+		$active_modules = $this->get_active_modules();
+
+		if ( empty( $active_modules ) ) {
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				__( 'Tag status not available: AdSense, Tag Manager, and Analytics modules are not connected.', 'google-site-kit' )
+			);
+
+			return $result;
+		}
+
+		$description = array();
+
+		foreach ( $active_modules as $module ) {
+			$tag_found = $this->check_if_tag_exists( $module->slug, $response );
+
+			if ( $tag_found ) {
+				$description[] = $tag_found;
+			}
+		}
+
+		if ( ! empty( $description ) ) {
+			$result['description'] = '<ul>' . join( "\n", $description ) . '</ul>';
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Gets active modules filtered to account only for
+	 * Analytics, AdSense and Tag Manager.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Filtered active modules instances.
+	 */
+	protected function get_active_modules() {
+		$active_modules = $this->modules->get_active_modules();
+
+		$active_modules = array_filter(
+			$active_modules,
+			function( $module ) {
+				return in_array(
+					$module->slug,
+					array(
+						Analytics_4::MODULE_SLUG,
+						AdSense::MODULE_SLUG,
+						Tag_Manager::MODULE_SLUG,
+					),
+					true
+				);
+			}
+		);
+
+		/**
+		 * Filtered list of active modules.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param array $active_modules An array of active module instances.
+		 */
+		return apply_filters( 'googlesitekit_site_status_get_active_modules', $active_modules );
+	}
+
+	/**
+	 * Checks if tag exists.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $module_slug Module slug.
+	 * @param string $content     Content to search for the tags.
+	 * @return bool TRUE if tag is found, FALSE if not.
+	 */
+	protected function check_if_tag_exists( $module_slug, $content ) {
+		if ( 'adsense' === $module_slug ) {
+			$module_name = 'AdSense';
+			$tag_matcher = new AdSense_Tag_Matchers();
+		}
+		if ( 'analytics-4' === $module_slug ) {
+			$module_name = 'Analytics';
+			$tag_matcher = new Analytics_Tag_Matchers();
+		}
+		if ( 'tagmanager' === $module_slug ) {
+			$module_name = 'Tag Manager';
+			$tag_matcher = new Tag_Manager_Tag_Matchers();
+		}
+
+		$search_string = 'Google ' . $module_name . ' snippet added by Site Kit';
+		/**
+		 * Filters the search string used to check if a specific Google service tag exists in the content.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param string $search_string The initial search string constructed for tag detection.
+		 * @param string $module_slug   The slug of the module whose tag is being checked.
+		 * @param string $content       The content in which the tag is being searched for.
+		 */
+		$search_string = apply_filters( 'googlesitekit_site_status_check_if_tag_exists', $search_string, $module_slug, $content );
```

Comment:
This filter is also not needed. Let's remove it too. We rarely add new filters, only if they are really needed.